src/Servers/Kestrel/Core/src/Internal/Http3/Http3Stream.cs

```diff
@@ -46,8 +47,7 @@ internal abstract partial class Http3Stream : HttpProtocol, IHttp3Stream, IHttpH
     private int _totalParsedHeaderSize;
     private bool _isMethodConnect;
 
-    // TODO: Change to resetable ValueTask source
-    private TaskCompletionSource? _appCompleted;
+    private readonly ManualResetValueTaskSource<object?> _appCompletedTaskSource = new ManualResetValueTaskSource<object?>();
```

Comment:
It's a _mutable struct_, so it shouldn't be `readonly`?
```suggestion
    private ManualResetValueTaskSource<object?> _appCompletedTaskSource = new ManualResetValueTaskSource<object?>();    // mutable struct, don't make readonly
```

Reply:
There's only one another IMO wrong use here:
https://github.com/dotnet/aspnetcore/blob/64c37113aafcbad73113b7cb86f86129e8839963/src/Servers/Kestrel/Core/src/Internal/Http2/Http2OutputProducer.cs#L37
`readonly` should be removed. All other uses here and in the runtime-repo are correct.

Reply:
🙈 wrong pattern matching by my eyes... Thanks.